gpt4all-bindings/typescript/index.cc

```diff
@@ -132,6 +132,11 @@ Napi::Value NodeModelWrapper::GetRequiredMemory(const Napi::CallbackInfo& info)
             library_path = ".";
         }
         device = config_object.Get("device").As<Napi::String>();
+        if(device != "cpu" && !(config_object.Has("nCtx") && config_object.Get("nCtx").IsNumber())){
+          throw Napi::Error::New(env,"nCtx is a required option when creating a model on the gpu");
+        }
```

Comment:
Why? nGpuLayers is the parameter that only applies when there is a GPU device. nCtx has the same effect regardless of device.

Reply:
Will look at, thanks

Reply:
should i just remove this check in general?

Reply:
Yeah, you could just remove it. The python bindings don't restrict any combination of device/n_ctx/ngl options, and they don't really need to - they are all well-defined, even if redundant.